topydo/ui/columns/Transaction.py

```diff
@@ -0,0 +1,76 @@
+# Topydo - A todo.txt client written in Python.
+# Copyright (C) 2017 Bram Schoenmakers <bram@topydo.org>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from topydo.lib.MultiCommand import MultiCommand
+
+
+class Transaction(object):
+    """
+    This class implements basic handling of executing any subcommand on multiple
+    todo items.
+    """
+    def __init__(self, p_subcommand=None, p_env_args=(), p_todo_ids=None):
+        self._multi = issubclass(p_subcommand, MultiCommand)
+        self._cmd = lambda op: p_subcommand(op, *p_env_args)
+        self._todo_ids = p_todo_ids
+        self._operations = []
+        self._cmd_name = p_subcommand.__module__.lower()[16:-7]
```

Comment:
> (...) perhaps it should be a method of the `Command` base class

Right now it seems so obvious and proper, that I'm embarassed that I didn't think of that myself. It will also solve the problem with `DeleteCommand` saving in backups as `delete` and therefore not paste-friendly (there is no such command as `delete`).